version: '3.7'

services:
  ntfy:
    # 2023-03-12
    image: binwiederhier/ntfy:v2.1.2@sha256:6a5db6e74d09d8fff91a557cedafb03e8817728e61ad27b61eaef294f299f79e
    environment:
      TZ: 'America/New_York'
    command:
      - "serve"
    ports:
      - "80"
    volumes:
      - "~/docker/services/ntfy/server.yml:/etc/ntfy/server.yml:ro"
      - "~/docker/data/ntfy/cache.db:/var/cache/ntfy/cache.db:rw"
      - "~/docker/data/ntfy/user.db:/var/lib/ntfy/user.db:rw"
    restart: unless-stopped
    depends_on:
      - traefik
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.ntfy.rule=Host(`ntfy.zah.rocks`)"
      - "traefik.http.routers.ntfy.entrypoints=websecure"
      - "traefik.http.routers.ntfy.tls.certresolver=myresolver"
